src/components/interface/nic/nic.h

```diff
@@ -4,8 +4,17 @@
 #include <cos_types.h>
 #include <cos_component.h>
 #include <cos_stubs.h>
+#include <shm_bm.h>
 
-int nic_send_packet(char* pkt, size_t pkt_size);
-int nic_bind_port(u32_t ip_addr, u16_t port, void* share_mem, size_t share_mem_sz);
+void nic_shmem_map(cbuf_t shm_id);
 
+int nic_send_packet(shm_bm_objid_t pktid, u16_t pkt_offset, u16_t pkt_len);
```

Comment:
This requires the client and server to manually do all of the conversions themselves for the shared memory, and means that *cannot* replace this shared memory implementation with another for this interface.

On Monday we need to talk about how to hide all of the mechanics behind the shared memory of the API, and make the interface simply pass pointers.

I'm fine with this PR currently, but this mistake cannot continue to pervade the system. The amount of refactoring in the future to fix it will be untenable.